### ISSUE TITLE
osproc: using `readLine` to read stdout of a program is > 100 times slower than in other languages

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#9975]
## TLDR

using `readLine` to read stdout of a program is > 100 times slower than in other languages

## timing
dummy child program that prints n lines to stdout:
nim c -d:release -d:case1 -o:/tmp/z10.dummy $timn_D/tests/nim/all/t0047.nim

master program that reads from child program:
nim c -d:release -d:case2 -o:/tmp/z10.master $timn_D/tests/nim/all/t0047.nim

ldmd2 -O -inline -release -noboundscheck -of/tmp/z09.2 $timn_D/tests/nim/all/t0048.d

time /tmp/z10.master '/tmp/z10.dummy 1000000'
1200000000
/tmp/z10.master '/tmp/z10.dummy 1000000'  70.39s user 0.69s system 101% cpu 1:10.09 total

time /tmp/z09.2 '/tmp/z10.dummy 1000000'
1200000000
/tmp/z09.2 '/tmp/z10.dummy 1000000'  0.57s user 0.31s system 170% cpu 0.515 total

## code

dummy child program and master program:
```nim
import os
import osproc
import streams
import strutils

proc dummyFun(n:int)=
  var line = ""
  for i in 0..<100:
    line.add "lorem ipsum "
  for i in 0..<n:
    stdout.writeLine line

proc mainFun(cmd: string)=
  # poParentStreams
  let p = startProcess(cmd, options = {poEvalCommand})
  defer: close(p)
  let sub = outputStream(p)
  var line: string
  var counter=0
  while true:
    let more = readLine(sub, line)
    counter.inc line.len
    # echo counter
    if not more: break
  echo counter

proc main()=
  when defined(case1):
    dummyFun(commandLineParams()[0].parseInt)
  when defined(case2):
    mainFun(commandLineParams()[0])
main()
```

D program that does same as master program:
```d
import std.stdio;
import std.process;
void main(string[]args){
  auto cmd = args[1];
  auto pipes = pipeShell(cmd, Redirect.stdout);
  scope(exit) wait(pipes.pid);
  size_t counter = 0;
  foreach (line; pipes.stdout.byLine){
    counter+=line.length;
  }
  writeln(counter);
}
```

## links
D20181214T015954

## notes
indeed, looks like the implementation of readLine is very inefficient right now, character by character

**Comment by narimiran** [comment]
Duplicate of https://github.com/nim-lang/Nim/issues/9026.
